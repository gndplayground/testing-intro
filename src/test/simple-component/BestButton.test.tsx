import React from "react";
import renderer from "react-test-renderer";
import BestButton from "./BestButton";

describe("Button", () => {
  it("Should render button", () => {
    const component = renderer.create(<BestButton>Hello</BestButton>);

    let tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("Should render link", () => {
    const component = renderer.create(
      <BestButton href="/URL">Hello</BestButton>
    );

    let tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("Should pass props", () => {
    const component = renderer.create(
      <BestButton aria-busy={true} onClick={() => {}}>
        Hello
      </BestButton>
    );

    let tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
