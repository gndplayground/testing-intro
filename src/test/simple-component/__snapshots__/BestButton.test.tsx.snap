// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Should pass props 1`] = `
<button
  aria-busy={true}
  className="my-button "
  onClick={[Function]}
>
  Hello
</button>
`;

exports[`Button Should render button 1`] = `
<button
  className="my-button "
>
  Hello
</button>
`;

exports[`Button Should render link 1`] = `
<a
  className="my-button my-button--link "
  href="/URL"
>
  Hello
</a>
`;
